Book-Your-Place/bookyourplace
# Implement trip view design

#### What's this PR do?
Implements trips#index view design.
Replaces the boilerplate scaffolding view with @stevepetebruce 's design.

Subsequent work will implement the user#show view design.

##### Background context
@stevepetebruce 's designs are now incorporated into the app, so now he can work in the app, not on standalone wireframes.

Many of the files added are just new image files, so the review shouldn't be so daunting.

There was some difficulty in adding some of the third-party vendor styles, esp. the "StarAdmin-Free-Bootstrap-Admin-Template" (original repo [here](https://github.com/BootstrapDash/StarAdmin-Free-Bootstrap-Admin-Template)). As the authors hadn't published a yarn package for their code, I had to, which is [here](https://yarnpkg.com/en/package/star-admin-free).

But once published and imported into this project there was an issue with the images in the styles.css. CSS that referenced `background: url(...` were not having those images resolved.

I had to work out a way to make this work with webpack. Multiple days playing around with webpack loaders (url-loader and file-loader) were fruitless, so I created a new version of the package with references to those images that just work with webpack without the need to use custom loaders. So the image references were:

`background: url("../images/samples/weather.svg")`

And are now:
`background: url("~/bootstrap-admin-dashboard-free/images/samples/weather.svg")`

(Feels a bit hacky but perhaps subsequent work can make the existing version and webpack play nicely without this change.)

The repo for the new bootstrap-admin-dashboard is [here](https://github.com/AlanDonohoe/admin-bootstrap-dashboard) and it's yarn package is [here]((https://yarnpkg.com/en/package/bootstrap-admin-dashboard-free)) (which uses these new webpack-friendly image references). So any updates to these styles, can be pushed up here and then publish the update to yarn also... then run (in this app):

`yarn upgrade bootstrap-admin-dashboard-free`

#### Where should the reviewer start?
Ignore the vast majority of the files (28 jpg images, etc) and start at:
app/javascript/packs/application.js and then app/views/*


#### How should this be manually tested?
Install all the dependencies (`yarn`) and then run the app (rails s / heroku local)... and make sure everything works.

#### Screenshots
(Still needs some tweaks but the main mechanics work)
![image](https://user-images.githubusercontent.com/1453680/45596103-549fb180-b9ae-11e8-9c5f-3493f816aaea.png)


---

#### Migrations
none

#### Additional deployment instructions
need to run `yarn` (to install dependencies)

#### Additional ENV Vars
none
